File: composite/java/TextFile.java
package androidhot.github.io.designpatterndemo.composite.java;

public class TextFile extends AbstractFile {
    private String fileName;

    public TextFile(String fileName) {
        this.fileName = fileName;
    }
    @Override public void add(AbstractFile abstractFile) {

    }


    @Override public void remove(AbstractFile abstractFile) {

    }


    @Override public AbstractFile getChild(int index) {
        return null;
    }


    @Override public void killVirus() {
        System.out.println("Kill file " + fileName + " virus.");
    }
}

File: composite/kotlin/TextFile.kt
package androidhot.github.io.designpatterndemo.composite.kotlin

class TextFile(private val fileName: String) : AbstractFile() {
    override fun add(abstractFile: AbstractFile) {}

    override fun remove(abstractFile: AbstractFile) {}

    override fun getChild(index: Int): AbstractFile? {
        return null
    }

    override fun killVirus() {
        println("Kill file $fileName virus.")
    }
}
